pagopa/io-app
# chore: [IA-802] Remove the button for (un)selecting all messages

## Short description
This PR aims to remove the button for selecting/unselecting all the messages, while in editing mode. Instead of simply removing it (as it might be useful in another context) I've made it optional.

| Old | New |
| - | - |
| ![Simulator Screen Shot - iPhone 13 Pro - 2022-04-04 at 14 41 39](https://user-images.githubusercontent.com/467098/161555515-648bd541-3264-4581-9b56-4ada4612ca1c.png) | ![Simulator Screen Shot - iPhone 13 Pro - 2022-04-04 at 14 41 14](https://user-images.githubusercontent.com/467098/161555563-1356fcad-2c6e-44e6-9ced-972b5eb1303f.png) |

## List of changes proposed in this pull request
- The `onToggleAllSelection` prop in `ListSelectionBar` is now optional. If passed to the component, the toggle all button is shown, otherwise it's not presented to the user.
- Styles have been adapted in order to adjust the layout in both the configurations (3 vs 2 buttons).
- Tests have been updated.

## How to test
Enable pagination and start selecting messages on both Inbox and Archive. Please beware that this modification has been made just for paginated messages, so in the old UI it should behave as usual.
